jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BuildFiles.java

```diff
@@ -98,4 +103,42 @@ public static JibContainerBuilder toJibContainerBuilder(
     }
     return containerBuilder;
   }
+
+  @VisibleForTesting
+  // TODO: add testing, need to do via intergration tests as there's no good way to extract out that
+  //   the base image was populated as the user intended currently.
+  static JibContainerBuilder createJibContainerBuilder(
+      BaseImageSpec from, JibCli buildOptions, ConsoleLogger logger)
+      throws InvalidImageReferenceException, FileNotFoundException {
+    String baseImageReference = from.getImage();
+    if (baseImageReference.startsWith(DOCKER_DAEMON_IMAGE_PREFIX)) {
+      return Jib.from(
+          DockerDaemonImage.named(baseImageReference.replaceFirst(DOCKER_DAEMON_IMAGE_PREFIX, "")));
+    }
+    if (baseImageReference.startsWith(TAR_IMAGE_PREFIX)) {
+      return Jib.from(
+          TarImage.at(Paths.get(baseImageReference.replaceFirst(TAR_IMAGE_PREFIX, ""))));
+    }
+    ImageReference imageReference =
+        ImageReference.parse(baseImageReference.replaceFirst(REGISTRY_IMAGE_PREFIX, ""));
+    RegistryImage registryImage = RegistryImage.named(imageReference);
+    DefaultCredentialRetrievers defaultCredentialRetrievers =
+        DefaultCredentialRetrievers.init(
+            CredentialRetrieverFactory.forImage(
+                imageReference,
+                logEvent -> logger.log(logEvent.getLevel(), logEvent.getMessage())));
+    Credentials.getFromCredentialRetrievers(buildOptions, defaultCredentialRetrievers)
+        .forEach(registryImage::addCredentialRetriever);
+    JibContainerBuilder containerBuilder = Jib.from(registryImage);
+    if (!from.getPlatforms().isEmpty()) {
+      containerBuilder.setPlatforms(
+          from.getPlatforms()
+              .stream()
+              .map(
+                  platformSpec ->
+                      new Platform(platformSpec.getArchitecture(), platformSpec.getOs()))
+              .collect(Collectors.toSet()));
```

Comment:
The iteration order is actually important. Internally we used `LinkedHashSet` everywhere to preserve the order while conveniently removing duplicates (also signifying that duplicates can't exist). However, I see that it can easily make the Jib API user to mess up with the order. Maybe accepting a `Set` wasn't a good idea.

Anyways, the question is, would `Collectors.toSet()` preserve the insertion order?

Reply:
no but I think I can do: `Collectors.toCollection(LinkedHashSet::new)`

Reply:
Why does the order matter? for selecting defaults?

Reply:
jib-core will create a manifest list with the given platform order.

Reply:
oh right.